test/files/pos/t9657-less-fishy.scala

```diff
@@ -0,0 +1,28 @@
+// scalac: -Werror
+
+// like pos/t9657
+// but with a more plausible reason for why to have a method type parameter
+
+sealed trait PowerSource
+sealed trait NonRenewablePowerSource extends PowerSource
+
+case object Petrol extends NonRenewablePowerSource
+case object Coal   extends NonRenewablePowerSource
+case object Pedal  extends PowerSource
+
+sealed abstract class Vehicle { type A <: PowerSource }
+case object Bicycle       extends Vehicle { type A = Pedal.type  }
+case class Bus(fuel: Int) extends Vehicle { type A = Petrol.type }
+case class Car(fuel: Int) extends Vehicle { type A = Petrol.type }
+
+object Test {
+  def refuel[P <: NonRenewablePowerSource](vehicle: Vehicle { type A = P }): Vehicle = vehicle match {
```

Comment:
Chatting with Fengyun, perhaps this is still fishy... why not just:

```scala
def refuel2(vehicle: Vehicle { type A <: NonRenewablePowerSource }): Vehicle = ...
```

?

Reply:
On this basis, I dropped this test case and, thus, the less-sure patch I had pushed.